tests/MongoDB.Driver.Core.Tests/Core/Configuration/ClusterBuilderTests.cs

```diff
@@ -59,10 +63,44 @@ public void CreateServerMonitorFactory_should_return_expected_result(int connect
 
             var serverSettings = result._settings();
         }
+
+        [Fact]
+        public void RegisterStreamFactory_should_complement_previous_RegisterStreamFactory_call()
+        {
+            string testValue = "TestValue";
+
+            var subject = new ClusterBuilder();
+            subject.RegisterStreamFactory(factory => new StreamFactoryTest { TestValue = testValue });
+            subject.RegisterStreamFactory(factory =>
+            {
+                var testFactory = (StreamFactoryTest)factory;
+                testFactory.TestValue.Should().Be(testValue);
+                return factory;
+            });
+            var streamFactoryWrapper = subject.GetStreamFactoryWrapperFunc();
+            streamFactoryWrapper(null);
```

Comment:
Can we come up with a better test? I think we want to test that all the wrappers were called in the correct order.